### ISSUE TITLE
MMStack is missing pages. 

### ISSUE THREAD

**Issue by manthasaigopal** [cabooster/SRDTrans#10]
I am trying to train the model on my own data and I am getting the following error. 
![Trainerror](https://github.com/cabooster/SRDTrans/assets/86771564/9e206a05-7943-4183-bcc3-306d1d50d599)

It says that certain number of pages are missing and also there is a division by zero error. Has it got anything to do with hyperstacks? How do I resolve the error. Each of the sample in my dataset has 45 frames. Awaiting your response.

**Comment by Huxiaowan** [comment]
Hello, our model is a 3D network that processes time-lapse imaging data. The input image size needs to be HWT, representing the image height, width, and time length respectively. It seems that the data format of each stack is four-dimensional. You need to modify the data and ensure that the number of time frames in the input stack is greater than "patch_t".

**Comment by manthasaigopal** [comment]
Yes! When there is only one image stack it is printing correct shape:
![image](https://github.com/cabooster/SRDTrans/assets/86771564/20511f46-41f7-48c7-baf1-46f4cb13d587)

But when there are more stacks then it is printing a 4 tupled vector: not sure why this is happening. I think the reason is that I am using MM stacks. But then how do I ensure that they are treated as separate tiff stacks

**Comment by cabooster** [comment]
I think your data may be a hyperstack. You can re-save the stack into a file using ImageJ. Then, the unnecessary channel will be removed. You can check the number of channels using ImageJ. I hope this is helpful for you.

**Comment by manthasaigopal** [comment]
Thanks for replying. I could resolve that issue and the code and running but the loss is very huge!!! 
![image](https://github.com/cabooster/SRDTrans/assets/86771564/bebf671c-96bd-4043-aa46-296869947917)

**Comment by manthasaigopal** [comment]
And also when I am using 126 stacks(each has 45 frames) for training with default parameters, I am getting division by zero error.

**Comment by cabooster** [comment]
This error is because the frames are too few to extract 3D training patches. I think a good idea to solve your problem is to duplicate your stack to form a larger stack. For example, you can duplicate your [1...45] (frames) stack into [1...45, 44...1, 2...45,...], just like image flip in t dimension. In this way, you can use a bigger patch_t to avoid unexpected problems.

**Comment by manthasaigopal** [comment]
That's a nice idea. Can you also comment about the loss!?

**Comment by cabooster** [comment]
I think the loss is OK. The scale of the loss will not affect training.

**Comment by manthasaigopal** [comment]
Will sampling rate of the frames affect the training(I did not get desired result)? If yes, then what is a reasonable sampling rate?
